gto-support-db/src/androidTest/java/org/ccci/gto/android/common/db/AbstractDaoTest.java

```diff
@@ -113,6 +112,70 @@ public void testHaving() {
         cursor.moveToFirst();
 
         assertThat(CursorUtils.getString(cursor, RootTable.COLUMN_TEST), is("2"));
+
+        cursor.close();
+    }
+
+    @Test
+    public void testCompoundKey() throws Exception {
```

Comment:
One thing to consider with this test in its current state is, if the test failed, we would have no idea which part of it failed without looking deeper.  If each piece was split out into a separate test method (create test, refresh test, update test, partial update test, conflict test, delete test) it would be more obvious.